static/src/javascripts/bootstraps/enhanced/football.js

```diff
@@ -0,0 +1,426 @@
+                        matchDayFor(competition, resp.matchDate)
+                            .fetch(container)
+                            .then(() => {
+                                extras[1] = {
+                                    name: 'Today’s matches',
+                                    importance: 2,
+                                    content: container,
+                                    ready: true,
+                                };
+                                renderExtras(extras, dropdownTemplate);
+                            })
+                            .catch(() => {
+                                delete extras[1];
+                                renderExtras(extras, dropdownTemplate);
+                            });
+                    });
+                });
+            });
+        }
+    });
+
+    isCompetition((competition: string) => {
+        const $rightHandCol = $('.js-secondary-column').dim().height;
+        if ($rightHandCol === 0 || $rightHandCol > 1800) {
+            renderTable(competition, extras, dropdownTemplate);
+        }
+    });
+
+    isFootballStatsPage(() => {
+        $('.js-chart').each(el => {
+            new TableDoughnut().render(el);
+        });
+    });
+
+    isLiveClockwatch(() => {
+        const isItACompetition = isCompetition();
+        const ml = new MatchListLive(
+            'match-day',
+            isItACompetition === false ? false : 'premierleague',
+            config.dateFromSlug()
+        );
+        const $img = $('.media-primary');
+        const $matchListContainer = $.create(
+            `
+            <div class="football-matches__container"
+                  data-link-name="football-matches-clockwatch"></div>
+        `
+        ).css({ minHeight: $img[0] ? $img[0].offsetHeight : 0 });
+
+        $img.addClass('u-h');
+
+        loading($matchListContainer[0], 'Fetching today’s matches…', {
+            text: 'Impatient?',
+            href: '/football/live',
+        });
+
+        $('.js-football-meta').append($matchListContainer);
+
+        const handleResponse = (success: boolean): void => {
+            if (
+                !success ||
+                $('.football-match', $matchListContainer[0]).length === 0
+            ) {
+                ml.destroy();
+                $matchListContainer.remove();
+                $img.removeClass('u-h');
+            }
+
+            $matchListContainer.css({ minHeight: 0 });
+            loaded($matchListContainer[0]);
+        };
+
+        ml
+            .fetch($matchListContainer[0])
+            .then(() => {
+                handleResponse(true);
+            })
+            .catch(() => {
+                handleResponse(false);
+            });
+    });
+
+    // Binding
+    bean.on(
+        document.body,
+        'click',
+        '.js-show-more-football-matches',
+        (e: Event): void => {
+            e.preventDefault();
+
+            const el =
+                e &&
```

Comment:
Probably don't this existential check (if we did, then `e.preventDefault()` would already have failed!)